Add caveat about Symbol.species being not recommended, new SAB[@@species] page (#20601)
mdn/content: files/en-us/web/javascript/reference/global_objects/array/@@species/index.md

The **`Array[@@species]`** accessor property returns the constructor used to construct return values from array methods.

> > **Warning:** The existence of `@@species` allows execution of arbitrary code and may create security vulnerabilities. It also makes certain optimizations much harder. Engine implementers are [investigating to remove this feature](https://github.com/tc39/proposal-rm-builtin-subclassing). Avoid relying on it if possible.

Reviewer: ```suggestion
> **Warning:** The existence of `@@species` allows execution of arbitrary code and may create security vulnerabilities. It also makes certain optimizations much harder. Engine implementers are [investigating whether to remove this feature](https://github.com/tc39/proposal-rm-builtin-subclassing). Avoid relying on it if possible.
```

### Species in ordinary objects

The `@@species` property returns the default constructor function, which is the `Array` constructor for `Array`.

```js
Array[Symbol.species]; // [Function: Array]
```

### Species in derived objects

> In a derived collection object (e.g. your custom array `MyArray`), the `MyArray` species is the `MyArray` constructor. However, you might want to overwrite this, in order to return parent `Array` objects in your derived class methods:

Reviewer: ```suggestion
In a derived collection object such as the custom array `MyArray`, the `MyArray` species is the `MyArray` constructor. However, you might want to overwrite this, in order to return parent `Array` objects in your derived class methods:
```
